src/components/Tool/FixCommitmentInfo/Contact.tsx

```diff
@@ -79,225 +126,477 @@ const useStyles = makeStyles()(() => ({
-      </Grid>
+      <Formik
+        initialValues={{
+          statusValue: statusValue,
+          pledgeCurrency: amountCurrency,
+          pledgeAmount: amount || suggestedAmount,
+          pledgeFrequency:
+            (frequencyValue as PledgeFrequencyEnum) || suggestedFrequency,
+        }}
+        validationSchema={commitmentInfoFormSchema}
+        onSubmit={async (values) => {
+          await onSubmit(values);
+        }}
+      >
+        {({
+          values: {
+            statusValue,
+            pledgeCurrency,
+            pledgeAmount,
+            pledgeFrequency,
+          },
+          handleSubmit,
+          setFieldValue,
+          errors,
+        }): ReactElement => {
+          const modalContact = {
+            id: id,
+            status: statusValue,
+            name: name,
+            pledgeCurrency,
+            pledgeAmount,
+            pledgeFrequency,
+          };
+          return (
+            <Form onSubmit={handleSubmit}>
+              <Grid container className={classes.formWrapper}>
+                <Card className={classes.formInner}>
+                  <Grid container>
+                    <Grid item lg={4} md={4} xs={12}>
+                      <Box
+                        display="flex"
+                        p={2}
+                        alignItems="center"
+                        className={classes.left}
+                      >
+                        <ContactAvatar
+                          src={avatar || ''}
+                          aria-label="Contact Avatar"
+                        />
+                        <Box display="flex" flexDirection="column" ml={2}>
+                          <Link
+                            data-testid="contactSelect"
+                            underline="hover"
+                            onClick={() => setContactFocus(id, 'Donations')}
+                          >
+                            <Typography variant="subtitle1">{name}</Typography>
+                          </Link>
+                          <Typography variant="subtitle2">
+                            {`Current: ${statusTitle} ${
+                              amount && amountCurrency
+                                ? currencyFormat(amount, amountCurrency, locale)
+                                : ''
+                            } ${getLocalizedPledgeFrequency(
+                              t,
+                              pledgeFrequency,
+                            )}`}
+                          </Typography>
+                        </Box>
+                      </Box>
+                    </Grid>
+                    <Grid item xs={12} md={8} lg={8} className={classes.right}>
+                      <Grid
+                        container
+                        style={{ paddingRight: theme.spacing(1) }}
+                      >
+                        <Grid item xs={12} md={6} lg={12}>
+                          <Box className={classes.boxTop}>
```

Comment:
For all these `Box`'s does it work to put the class on the parent `Grid` and remove the `Box` so that it's one fewer level of nesting? That's an optional suggestion since that element structure was already there.